src/target/stm32f1.c

```diff
@@ -486,9 +499,14 @@ static bool stm32f1_cmd_option(target *t, int argc, const char **argv)
 	target_mem_write32(t, FLASH_OPTKEYR, KEY1);
 	target_mem_write32(t, FLASH_OPTKEYR, KEY2);
 
-	if ((argc == 2) && !strcmp(argv[1], "erase")) {
+	if ((argc == 2) && strcmp(argv[1], "erase") == 0) {
 		stm32f1_option_erase(t);
-		stm32f1_option_write_erased(t, FLASH_OBP_RDP, flash_obp_rdp_key);
+		bool width_word = false;
+		if (strcmp(t->driver, "GD32E230") == 0) {
```

Comment:
If possible, can we detect this based on, eg, t->idcode? This works but we'd like a cheaper option that uses a bit less Flash and time